### ISSUE TITLE
Check stack count before opening shulker box

### ISSUE THREAD

**Issue by ApliNi** [percyqaz/Shulkerbox#1]
Some servers make empty shulker boxes stackable via stacking plugins, need to check the stack count before opening them.

I do this in other plugins
```java
public void onClick(InventoryClickEvent e) {
    ItemStack clicked = e.getCurrentItem();
    // Do not open if stacked: compatible stacking plugin
    if(clicked != null && clicked.getAmount() != 1) return;

    // ...
}
```

**Comment by percyqaz** [comment]
Thanks for the request
I think the fix will not be just as simple as checking the stack count

What happens if someone stacks empty shulker boxes while this plugin has an open shulker box?
Changes are only saved to the stack when the inventory gets closed so I think you would be able to get a duplication bug by:
- Opening an empty shulker box
- Putting items in it
- Stacking the shulker box with other empty boxes
- Close shulker box (my plugin will now save the contents to every shulker box in the stack)
- You have duplicated items

I hope that makes sense, what do you think about this?
Also, if you have a plugin to hand that allows empty shulker stacking can you send it to me as that means I can do thorough testing with both plugins installed

**Comment by ApliNi** [comment]
I think, you fixed it, other shulker boxes cannot be left-clicked while opening a shulker box, but this leads me to another possibility: the player picks up an empty shulker box item after opening , they will be merged

plugin: https://www.spigotmc.org/resources/simple-stack-stack-any-items-to-64.83044/
Please add shulker boxes using the Unique Iten List feature of this plugin, in the config GUI

**Comment by percyqaz** [comment]
Yes, setting random NBT data on the shulker box should prevent it from stacking with other boxes and causing issues

So the goal is:
- The instant a shulker box is right clicked, ensure it is a stack of 1 + mark it with random NBT
- The instant a shulker box is closed (including if the player dies or disconnects), ensure the NBT is removed

Now during a shulker box being opened it should be impossible for identical shulker boxes to stack with it

Is there a way this could fail if the server crashes while someone has a shulker box open and this leaves the NBT tag on the box forever? If so, should we just override that data when it's next opened and that will fix any glitched boxes?

Also, do you know a good way to set NBT data on item stacks without using `net.minecraft.server`?
I would prefer not to use it as then the plugin will only target a specific version of Spigot and won't be future proof.
I'm not a very experienced plugin developer so I wouldn't have the time to set up a system to update the plugin for each version of NMS/Spigot

**Comment by ApliNi** [comment]
I am studying how to operate nbt, there may be a better nbt library.

I may update the code on another branch: https://github.com/ApliNi/Shulkerbox
